src/Apps/Auction/Routes/ConfirmBid/index.tsx

```diff
@@ -90,26 +108,51 @@ export const ConfirmBidRoute: React.FC<BidProps> = props => {
       errorMessages = [error.message]
     }
 
-    // TODO: add tracking with errorMessages
-    logger.error(errorMessages) // remove once tracking is implemented
+    trackConfirmBidFailed(bidderId, errorMessages)
 
     actions.setSubmitting(false)
     actions.setStatus("submissionFailed")
   }
 
+  function trackConfirmBidFailed(bidderId: string, errors: string[]) {
+    tracking.trackEvent({
+      action_type: Schema.ActionType.ConfirmBidFailed,
+      bidder_id: bidderId,
+      error_messages: errors,
+      ...commonProperties,
+    })
+  }
+
+  function trackConfirmBidSuccess(positionId: string, bidderId: string) {
+    tracking.trackEvent({
+      action_type: Schema.ActionType.ConfirmBidSubmitted,
+      bidder_position_id: positionId,
+      bidder_id: bidderId,
+      ...commonProperties,
+    })
+  }
+
   function handleSubmit(
     values: { selectedBid: number },
     actions: FormikActions<object>
   ) {
+    const bidderId = sale.registrationStatus.id
```

Comment:
Just want to surface that it's possible for the underlying gravity endpoint, [`POST /api/v1/me/bidder_position`](https://github.com/artsy/gravity/blob/b38e5a5dbf5c181f9ab3b86a34091817a7f6a03d/app/api/v1/me_bidder_positions_endpoint.rb#L64-L68), to _create_ a `Bidder` document in Gravity's mongo, so is it safe to assume that we'll always have a bidder at this point?

Maybe for the "just bid" flow, instead of the "bid+register" flow, we'll feel free confident that we always have a preexisting `Bidder` document, but I think we'll have to re-visit this when implementing the "bid+register" flow

Did you guys discuss this during implementation?